src/game.cpp

```diff
@@ -6400,12 +6400,23 @@ void game::examine( const tripoint &examp )
 //represents carefully peeking around a corner, hence the large move cost.
 void game::peek()
 {
-    const cata::optional<tripoint> p = choose_adjacent( _( "Peek where?" ) );
+    const cata::optional<tripoint> p = choose_adjacent( _( "Peek where?" ), true );
     if( !p ) {
         refresh_all();
         return;
     }
 
+    if( ( *p ).z != 0 ) {
```

Comment:
```suggestion
    if( p->z != 0 ) {
```
Same for all the code below.